realm/realm-library/src/main/cpp/io_realm_internal_CheckedRow.cpp

```diff
@@ -24,214 +27,213 @@ using namespace realm;
 JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnCount
   (JNIEnv* env, jobject obj, jlong nativeRowPtr)
 {
-    if (!ROW(nativeRowPtr)->is_attached())
-        return 0;
+    return try_catch<jlong>(env, [&]() {
+        Row* row = reinterpret_cast<Row*>(nativeRowPtr);
+        if (!row->is_attached())
+            return static_cast<jlong>(0);
```

Comment:
Because of the template. `0` is `int` and we return a `jlong` also. The compiler will not allow multiple return types (and gcc cannot figure out that 0 also can be a `jlong`).